### PR TITLE
Add spellcheck

### PR DESCRIPTION
**Pull request by reyang**
1. add spell check to docs and code
2. remove stale TBDs

Open question: do we think GitHub Actions are okay or we want to restrict to Circle CI?

### CODE REVIEW

File: .github/workflows/spellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        sh ./install-misspell.sh
+
+    - name: run misspell
+      run: |
+        ./bin/misspell -error ./**/*.h
+        ./bin/misspell -error ./**/*.cc
+        ./bin/misspell -error ./**/*.cmd
+        ./bin/misspell -error ./**/*.md
+        ./bin/misspell -error ./**/*.sh
+        ./bin/misspell -error ./**/*.txt
```
Comment (g-easy) [inline .github/workflows/spellcheck.yml]: This covers CMakeLists.txt. Should we also cover BUILD files?

Comment (reyang) [reply .github/workflows/spellcheck.yml]: Sure. Do you think we should just run against `*` (so we don't have to maintain a list here)?

Comment (g-easy) [reply .github/workflows/spellcheck.yml]: Dunno. :) Your call.

File: .github/workflows/spellcheck.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: check out code
+      uses: actions/checkout@v2
+
+    - name: install misspell
+      run: |
+        curl -L -o ./install-misspell.sh https://git.io/misspell
```
Comment (g-easy) [inline .github/workflows/spellcheck.yml]: Is there a less risky way to install this than `curl http://redirection-service | bash` ? :)

e.g. For autoformatting we "go get" a thing.

Comment (lalitb) [reply .github/workflows/spellcheck.yml:19]: There is option using

$ go get -u github.com/client9/misspell/cmd/misspell

but  don't think golang is part of default `ubuntu-latest` installation.

**Pull request closed without merging**